tests/FakeItEasy.Analyzer.Tests/NonVirtualSetupAnalyzerTests.cs

```diff
@@ -0,0 +1,263 @@
+﻿namespace FakeItEasy.Analyzer.Tests
+{
+    using FakeItEasy.Analyzer.Tests.Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Xunit;
+
+    public class NonVirtualSetupAnalyzerTests:DiagnosticVerifier
+    {
+        //The aim to create an analyzer that picks up
+        //on any call setups that target a non-virtual member
+
+        //A.CallTo(() => shop.GetTopSellingCandy()).Returns(lollipop); //Warning here
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Interface()
+        {
+            const string test = @"using FakeItEasy;
+namespace TheNamespace
+{
+    class TheClass
+    {
+        void Test()
+        {
+            var foo = A.Fake<IFoo>();
+            A.CallTo(() => foo.Bar()).Returns(42);
+        }
+    }
+
+    interface IFoo { int Bar(); }
+}
+";
+
+            VerifyCSharpDiagnostic(test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Virtual_Member()
+        {
+            const string test = @"using System;
+
+//Temporary class to get non-virtual test correct
+namespace FakeItEasy.Analyzer.Tests
+{
+    class TheClass
+    {
+        void TheTest()
+        {
+            var foo = A.Fake<Foo>();
+            A.CallTo(() => foo.Bar());
+        }
+    }
+
+    internal class Foo
+    {
+        internal virtual void Bar()
+        {
+            throw new NotImplementedException();
+        }
+    }

```

Comment:
Excellent point, @thomaslevesque. In fact, I'd upgrade this from "nitpick". The "test code" should be an otherwise-valid use of FakeItEasy.